### PR TITLE
Ver/6.4.2

### PR DESCRIPTION
**Pull request by duplexsystem**
Terra Version 6.4.2

- 1.20.3/4
- bring minimum supported minecraft version to 1.20.[3/4]
- enforce NMS bindings 
- fix CME
- additional warning if not using paper(+) as we super depend on it now
- minor git ignore changes
- small perf opt

### CODE REVIEW

File: platforms/bukkit/common/src/main/java/com/dfsek/terra/bukkit/handles/BukkitWorldHandle.java [hunk 1]
```diff
@@ -51,6 +61,13 @@ public BukkitWorldHandle() {
 
     @Override
     public @NotNull EntityType getEntity(@NotNull String id) {
+        if (!id.contains(":")) { //TODO: remove in 7.0
+            String newid = "minecraft:" + id.toLowerCase();;
+            logger.warn(
+                "Translating " + id + " to " + newid + ". In 1.20.3 entity parsing was reworked" +
+                ". You are advised to preform this rename in your config backs as this translation will be removed in the next major " +
```
Comment (Astrashh) [inline platforms/bukkit/common/src/main/java/com/dfsek/terra/bukkit/handles/BukkitWorldHandle.java:68]: preform typo

**Review by Astrashh — APPROVED** [review]
Mostly platform stuff so I don't have too much to add

File: platforms/fabric/src/main/resources/fabric.mod.json
Comment (justaureus) [inline platforms/fabric/src/main/resources/fabric.mod.json:1]: Is the minimum version now 1.20.4?  I thought for some reason we were targeting 1.20.2+

**Review by justaureus — APPROVED** [review]
Looks good overall! Just had the one question about the minimum version. :)

**Pull request merged**